prov/efa/src/efa.h

```diff
@@ -558,6 +558,10 @@ struct rdm_peer *rxr_ep_get_peer(struct rxr_ep *ep, fi_addr_t addr)
 {
 	struct util_av_entry *util_av_entry;
 	struct efa_av_entry *av_entry;
+
+	if (OFI_UNLIKELY(addr == FI_ADDR_NOTAVAIL))
```

Comment:
Might as well add the assert inside here instead of returning NULL. Trying to get the peer object without an address is always a programming error and points to missing logic at the caller. Also the next person to call `rxr_ep_get_peer()` doesn't have to remember to add an assert :)

Reply:
assert() just takes effect in debug mode. I think it's still needed to return NULL here to avoid segfault due to the call to ofi_bufpool_get_ibuf.